File: src/main/java/com/lixiang/car/decodecohttp/util/CodeManager.java
package com.lixiang.car.decodecohttp.util;

import com.intellij.openapi.project.Project;
import com.lixiang.car.decodecohttp.decodecohttp.util.NotifyUtil;

import java.io.File;
import java.util.UUID;

/**
 * @author shuzijun
 */
public class CodeManager {

    @Deprecated
    public static void openCode(Project project, String text) {
        openCode(project, text, "json-res-"+ UUID.randomUUID() +".json");
    }

    public static void openCode(Project project, String text, String fileName) {
        String filePath = FileUtils.defaultFileFolder() + File.separator + "decodeCohttp" + File.separator + fileName;

        File file = new File(filePath);
        if (file.exists()) {
            file.delete();
        }
        try {
            FileUtils.saveFile(file, text);
            FileUtils.openFileEditorAndSaveState(file, project);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static boolean saveAndOpen(Project project, String path, String text, String fileName) {
        String filePath = path + File.separator + fileName;

        File file = new File(filePath);
        if (file.exists()) {
            return false;
        }
        try {
            FileUtils.saveFile(file, text);
            FileUtils.openFileEditorAndSaveState(file, project);
        } catch (Exception e) {
            e.printStackTrace();
            NotifyUtil.INSTANCE.notifyMessage(e.getCause().toString());
        }
        return true;
    }
}

File: src/main/kotlin/com/lixiang/car/happytools/tools/util/CodeManager.kt
package com.lixiang.car.happytools.tools.util

import com.intellij.openapi.project.Project
import com.lixiang.car.happytools.tools.util.FileUtils.defaultFileFolder
import com.lixiang.car.happytools.tools.util.FileUtils.openFileEditorAndSaveState
import com.lixiang.car.happytools.tools.util.FileUtils.saveFile
import com.lixiang.car.happytools.tools.util.NotifyUtil.notifyMessage
import java.io.File
import java.util.*

object CodeManager {
    @Deprecated("")
    fun openCode(project: Project?, text: String?) {
        openCode(project, text, "json-res-" + UUID.randomUUID() + ".json")
    }

    fun openCode(project: Project?, text: String?, fileName: String) {
        val filePath = defaultFileFolder() + fileName
        val file = File(filePath)
        if (file.exists()) {
            file.delete()
        }
        try {
            saveFile(file, text)
            openFileEditorAndSaveState(file, project)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun onlyOpenCode(project: Project?, fileName: String, defaultContent: String?=null): Boolean {
        val filePath = defaultFileFolder() + File.separator + "decodeCohttp" + File.separator + fileName
        val file = File(filePath)
        try {
            if (!file.exists()) {
                saveFile(file, defaultContent)
            }
            openFileEditorAndSaveState(file, project)
        } catch (e: Exception) {
            e.printStackTrace()
            return false
        }
        return true
    }

    fun saveAndOpen(project: Project?, path: String, text: String?, fileName: String): Boolean {
        val filePath = path + File.separator + fileName
        val file = File(filePath)
        if (file.exists()) {
            return false
        }
        try {
            saveFile(file, text)
            openFileEditorAndSaveState(file, project)
        } catch (e: Exception) {
            e.printStackTrace()
            notifyMessage(e.cause.toString())
        }
        return true
    }
}